runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientSaslHandshaker.java

```diff
@@ -430,8 +568,98 @@ protected final int decodeSaslAuthenticate(
                 client.decodableResponseBytes -= authenticateResponse.sizeof();
                 assert client.decodableResponseBytes >= 0;
 
-                client.onDecodeSaslResponse(traceId);
-                client.onDecodeSaslAuthenticateResponse(traceId, authorization, errorCode);
+                if (client.sasl != null && client.sasl.mechanism != null &&
```

Comment:
Let's also split `decodeSaslAuthenticate` into `decodeSaslPlainAuthenticate`, `decodeSaslScramAuthenticateFirst` and `decodeSaslScramAuthenticateFinal` to follow the state machine rather than needing to have if conditions inside a general purpose `decodeSaslAuthenticate`.

Note: as with encode, we may need a further `decodeSaslAuthenticate` field to delegate to one of `decodeSaslPlainAuthenticate`, `decodeSaslScramAuthenticateFirst` or `decodeSaslScramAuthenticateFinal` by setting that `decodeSaslAuthenticate` value accordingly while following the encoding and decoding state machine.